src/mlpack/core/hpt/hpt.hpp

```diff
@@ -0,0 +1,268 @@
+/**
+ * @file hpt.hpp
+ * @author Kirill Mishchenko
+ *
+ * Hyper-parameter tuning.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_HPT_HPT_HPP
+#define MLPACK_CORE_HPT_HPT_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+#include <mlpack/core/hpt/deduce_hp_types.hpp>
+#include <mlpack/core/optimizers/grid_search/grid_search.hpp>
+
+namespace mlpack {
+namespace hpt {
+
+/**
+ * The class HyperParameterTuner for the given MLAlgorithm utilizes the provided
+ * Optimizer to find the values of hyper-parameters that optimize the value of
+ * the given Metric. The value of the Metric is calculated by performing
+ * cross-validation with the provided cross-validation strategy.
+ *
+ * To construct a HyperParameterTuner object you need to pass the same arguments
+ * as for construction of an object of the given CV class. For example, we can
+ * use the following code to try to find a good lambda value for
+ * LinearRegression.
```

Comment:
You're right, sorry about that.  Maybe I was looking at some branch or old version that did not have `lambda`.